syft/serde/torch_serde.py

```diff
@@ -326,30 +310,25 @@ def _detail_torch_parameter(worker: AbstractWorker, param_tuple: tuple) -> torch
     return param
 
 
-def _simplify_torch_device(worker: AbstractWorker, device: torch.device) -> Tuple[str]:
-    return device.type
-
+def _simplify_torch_device(worker: AbstractWorker, device: torch.device) -> Tuple:
+    device_type = syft.serde._simplify(worker, device.type)
+    return (device_type,)
 
-def _detail_torch_device(worker: AbstractWorker, device_type: str) -> torch.device:
-    return torch.device(type=device_type)
 
+def _detail_torch_device(worker: AbstractWorker, device_type: tuple) -> torch.device:
+    return torch.device(type=syft.serde._detail(worker, device_type[0]))
 
-def _simplify_torch_size(worker: AbstractWorker, shape: torch.Size) -> Tuple:
-    return (list(shape),)
 
-
-def _detail_torch_size(worker: AbstractWorker, shape: List[int]) -> torch.Size:
-    return torch.Size(*shape)
-
-
-def _simplify_script_module(worker: AbstractWorker, obj: torch.jit.ScriptModule) -> str:
+def _simplify_script_module(worker: AbstractWorker, obj: torch.jit.ScriptModule) -> Tuple:
     """Strategy to serialize a script module using Torch.jit"""
-    return obj.save_to_buffer()
+    return (obj.save_to_buffer(),)
 
 
-def _detail_script_module(worker: AbstractWorker, script_module_bin: str) -> torch.jit.ScriptModule:
+def _detail_script_module(
+    worker: AbstractWorker, script_module_bin: Tuple
+) -> torch.jit.ScriptModule:
     """"Strategy to deserialize a binary input using Torch load"""
-    script_module_stream = io.BytesIO(script_module_bin)
+    script_module_stream = io.BytesIO(script_module_bin[0])
```

Comment:
Q: We should get here with a ```len(script_module_bin)``` of size 1, right? I am thinking if we should add an assert of not.

Reply:
Yes. I haven't seen asserts in similar cases, e.g. in `_detail_str`, `_detail_range`, `_detail_ndarray`, etc.
Do you think they should be added?

Reply:
I think it is ok how it is now.